### DISCUSSION TITLE
QR is valid more than once

### DISCUSSION THREAD

**Discussion by JanineKall** [liedekef/events-made-easy#180] [Q&A]
Hi, I want to use QR codes in my events. I set up everything and it looks good, except for one thing: even as an authorized user I can allow a one-seat booking more than once. Every time I scan the code, an attendance record is added to the reservation, but scanning the codes still returns (in Dutch) 'aantal keer gescand=0, max aantal =1'. 
Any suggestions?
![image](https://github.com/liedekef/events-made-easy/assets/129589814/2633b5d5-bb75-4e4b-8fa7-eb5b42a73edb)
![Afbeelding van WhatsApp op 2023-05-21 om 14 55 56](https://github.com/liedekef/events-made-easy/assets/129589814/3e7d8299-624e-48dd-9d50-7967db7d95fd)

**Comment by liedekef** [comment]
I just tested this and it works as expected. The only possible explanation is that you have a caching engine of some kind. Is there something like that installed?

**Comment by JanineKall** [comment]
Not that I'm aware of. 
These are the plugins of my client's website: 
@Quest WordPress Onderhoud
Cornerstone 
Disable Comments 
Easy WP SMTP 
Email Address Encoder 
Events Made Easy 
Google Analytics 
Gravity Forms 
Gravity Forms (nl) 
Pronamic Pay (not active)
PublishPress Capabilities 
Redirection 
Under Construction (not active)
WP Test Email 
Yoast Duplicate Post 
Yoast SEO selecteren

**Comment by liedekef** [comment]
You can see the code in eme_events.php:

```
                        eme_update_attendance_count( $booking_id );
                        $attendance_count = eme_get_attendance_count( $booking_id );
```
Meaning: if that var is 0, the function call eme_update_attendance_count failed. And that function only does a simple "+1" (see the function definition).
I don't check for the return code (I could), but if that then stays 0, something fishy is going on.

**Comment by JanineKall** [comment]
Thanks for the answer, I see how it works, I suppose that means something sticky is going on. I can see the adding of the attendance record, but scanning the QR-code still returns 0. 
I have no access to my client's server at this point, is there something I can do?

**Reply by liedekef** [reply]
Via some SQL plugin, you could try to execute the sql I do in the function eme_update_attendance_count for a certain booking id, and see what happens.

**Reply by liedekef** [reply]
Also, this change will trap future attendance count update issues: https://github.com/liedekef/events-made-easy/commit/f05cb2299e58db68962a03ac4a365bb827bf1486

**Reply by JanineKall** [reply]
Oh I'm not familiar with such plug-ins. Which keywords can I use to find them in wordpress.org?

**Reply by liedekef** [reply]
phpmyadmin can help, but this is outside of EME support :-)
